### PR TITLE
[feature]add arrow type for streamload

### PR DESCRIPTION
**Pull request by wuwenchi**
# Proposed changes

add arrow type for streamload

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/serializer/RowDataSerializer.java [hunk 1]
```diff
@@ -40,12 +53,18 @@
  * Serializer for RowData.
  */
 public class RowDataSerializer implements DorisRecordSerializer<RowData> {
```
Comment (JNSimba) [inline flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/serializer/RowDataSerializer.java:55]: It is recommended to write an arrow ut in TestRowDataSerializer

**Comment by JNSimba** [comment]
The `writeRecord` method of `DorisStreamLoad` or `DorisBatchStreamload` will splice `lineDelimiter` characters (`\n`). Will it have any impact when arrow format writing is enabled?

**Comment by JNSimba** [comment]
Only RowDataSerializer has been modified. Is it because other Serializers cannot obtain the written data type?

File: flink-doris-connector/pom.xml [hunk 2]
```diff
@@ -124,6 +124,13 @@ under the License.
             <scope>provided</scope>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-python</artifactId>
+            <version>${flink.version}</version>
+            <scope>provided</scope>
```
[the remark this answers is not in the document]
Comment (wuwenchi) [reply flink-doris-connector/pom.xml:131]: In order to use ArrowSerializer. This class provides conversion from rowdata schema to arrow schema, and provides conversion from rowdata to arrow data.

**Comment by wuwenchi** [comment]
> The `writeRecord` method of `DorisStreamLoad` or `DorisBatchStreamload` will splice `lineDelimiter` characters (`\n`). Will it have any impact when arrow format writing is enabled?
Will not be affected because `lineDelimiter` will be set to null when using arrow.

**Comment by wuwenchi** [comment]
> Only RowDataSerializer has been modified. Is it because other Serializers cannot obtain the written data type?

Yes

File: flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/serializer/RowDataSerializer.java (hunk 1 above)
Comment (wuwenchi) [reply flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/serializer/RowDataSerializer.java:51]: done

File: flink-doris-connector/build.sh [hunk 3]
```diff
@@ -160,7 +162,7 @@ FLINK_MAJOR_VERSION=0
 echo_g " flink version: ${FLINK_VERSION}, major version: ${FLINK_MAJOR_VERSION}"
 echo_g " build starting..."
 
-${MVN_BIN} clean package -Dflink.version=${FLINK_VERSION} -Dflink.major.version=${FLINK_MAJOR_VERSION} "$@"
+${MVN_BIN} clean package -Dflink.version=${FLINK_VERSION} -Dflink.major.version=${FLINK_MAJOR_VERSION} -Dflink.python.id=${FLINK_PYTHON_ID} "$@"
```
Comment (JNSimba) [inline flink-doris-connector/build.sh:165]: This also needs to be added to the configuration items of the pipeline.

**Review by JNSimba — APPROVED** [review]
LGTM

**Pull request merged**